File: src/Java/Problem2.java
public class LRUCache {
    private final int capacity;
    HashMap<Integer,Integer> map = new HashMap<>();
    LinkedList<Integer> list = new LinkedList<>();

    public LRUCache(int capacity) {
        this.capacity = capacity;
    }

    public int get(int key) {
        if(this.map.containsKey(key)){
            this.moveCacheElements(key);
            return this.map.get(key);
        }
        return -1;
    }

    public void put(int key, int value) {

        if(this.map.containsKey(key)){
            this.moveCacheElements(key);
        }

        if(this.map.size() >= this.capacity){
            int removedElement = this.list.removeLast();
            this.map.remove(removedElement);
        }

        this.list.addFirst(key);
        this.map.put(key, value);

    }
    
    private void moveCacheElements(int key){
        this.list.remove((Integer) key);
        this.list.addFirst(key);
    }
}

File: src/Python/Problem2.py
from collections import deque

class LRUCache:
    def __init__(self, capacity: int):
        self.capacity = capacity
        self.list = deque()
        self.map = {}

    def get(self, key: int) -> int:
        if key in self.map:
            self.moveCacheElements(key)
            return self.map[key]
        return -1

    def put(self, key: int, value: int) -> None:
        if key in self.map:
            self.moveCacheElements(key)
        else:
            if len(self.map) >= self.capacity:
                removedElement = self.list.pop()
                self.map.pop(removedElement)

            self.list.appendleft(key) 
            self.map[key] = value

    def moveCacheElements(self, key) -> None:
        self.list.remove(key)
        self.list.appendleft(key)